### PR TITLE
feat: asyncApi files-scenarios link

### LINKED ISSUE #22: Create the link between asyncApi spec file and simPlot file
There needs to be an extension for the async-api file so that you can link routes from the async api file with corresponding Plots which are going to define details about how the application simulation should occur.

Status: working on it

### PR DESCRIPTION
**Pull request by NektariosFifes**
**Description**

- Create association between asyncApi file and scenario file in order to aquire and structure all information needed to 
send and event to an endpoint.

Tested handling of:
async api and scenario file with wrong  naming (only yaml and json files)
scenario file with wrong format not supported by the schema of the specified version
non existend asyncApi and scenario files

**Related issue(s)**
fixes #22

### CODE REVIEW

**Comment by NektariosFifes** [comment]
@jonaslagoni

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just some general comments and suggestions 🙂 👍 

Be very careful about creating function names such as `verifyInput_ParseAndLinkFiles` as they can grow undesirably long 😄

File: src/parser/scenarioParserAndConnector.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+ * that can be executed.
+ *
+ *
+ * @constructor
+ * @param filepathAsyncApi
+ * @param filepathScenario
+ */
+const scenarioParserAndConnector  = async (filepathAsyncApi, filepathScenario) => {
+  const parserContext = this;
+  parserContext.ready = false;
```
[the remark this answers is not in the document]
Comment (NektariosFifes) [reply src/parser/scenarioParserAndConnector.js:80]: its just a leftover from smth else i had done previously i will remove it

File: src/parser/scenarioParserAndConnector.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ * Asynchronously parses the provided YAML or JSON file, creates object containing details about the operations
+ * that can be executed.
+ *
+ *
+ * @constructor
+ * @param filepathAsyncApi
+ * @param filepathScenario
+ */
+const scenarioParserAndConnector  = async (filepathAsyncApi, filepathScenario) => {
+  const parserContext = this;
```
[the remark this answers is not in the document]
Comment (NektariosFifes) [reply src/parser/scenarioParserAndConnector.js:79]: i was just for declaring the variable at the start of the function and implying the role of the variable but it is not needed. i will remove

**Review by jonaslagoni — APPROVED** [review]
👍

**Pull request merged**